### PR TITLE
Renamed ambiguous Ray easyconfigs to Ray-assembler and Ray-project

### LINKED ISSUE #10757: Name collision with easyconfigs for Ray
We have two easyconfigs with the same name for two unrelated applications:
* `Ray-0.8.4-foss-2019b-Python-3.7.4.eb` for https://github.com/ray-project/ray
* `Ray-2.3.1-iimpi-2019a.eb` for https://github.com/sebhtml/ray

These packages should have different names to be able to load them at the same time.

I propose to change the name of both packages to avoid wrong assumptions from the users:
* `Ray-0.8.4-foss-2019b-Python-3.7.4.eb` could be renamed to Ray-Framework or Ray-Project
*  `Ray-2.3.1-iimpi-2019a.eb` could be renamed to Ray-Assembler or Ray-DenovoAssembler

### PR DESCRIPTION
**Pull request by Darkless012**
Renamed ambiguous Ray easyconfigs to be distinguishable.
More info in issue #10757

### CODE REVIEW

**Comment by smoors** [comment]
@Darkless012 thanks for doing this.
however, I'm not sure if it's a good idea to modify files in the archive since, to me, an archive is a sort of historical reference.
what do you think?

**Comment by Micket** [comment]
I don't mind changing the archive so that it matches the name of non-archived configs.  We have git history for actual history?

File: easybuild/easyconfigs/r/Ray-project/Ray-project-0.8.4-foss-2019b-Python-3.7.4.eb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 easyblock = 'PythonBundle'
 
-name = 'Ray'
+name = 'Ray-project'
```
Comment (smoors) [inline easybuild/easyconfigs/r/Ray-project/Ray-project-0.8.4-foss-2019b-Python-3.7.4.eb:6]: please adapt `source_tmpl` (below) to account for this change.

File: easybuild/easyconfigs/r/Ray-assembler/Ray-assembler-2.3.1-iimpi-2019a.eb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 easyblock = "MakeCp"
 
-name = "Ray"
+name = "Ray-assembler"
```
Comment (smoors) [inline easybuild/easyconfigs/r/Ray-assembler/Ray-assembler-2.3.1-iimpi-2019a.eb:6]: please adapt `sources` to account for this change.

File: easybuild/easyconfigs/__archive__/r/Ray-assembler/Ray-assembler-2.3.1-intel-2015a.eb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 easyblock = 'ConfigureMake'
 
-name = 'Ray'
+name = "Ray-assembler"
```
Comment (smoors) [inline easybuild/easyconfigs/__archive__/r/Ray-assembler/Ray-assembler-2.3.1-intel-2015a.eb:3]: please adapt sources to account for this change.

**Review by smoors — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
There's little point to change archived easyconfigs, the whole reason we've archived them is because we won't maintain them (and they'll be removed in the next major release of EasyBuild).

File: easybuild/easyconfigs/r/Ray-project/Ray-project-0.8.4-foss-2019b-Python-3.7.4.eb (hunk 1 above)
Comment (Darkless012) [reply easybuild/easyconfigs/r/Ray-project/Ray-project-0.8.4-foss-2019b-Python-3.7.4.eb:6]: Ah, thanks for the notice :)

File: easybuild/easyconfigs/r/Ray-assembler/Ray-assembler-2.3.1-iimpi-2019a.eb (hunk 2 above)
Comment (Darkless012) [reply easybuild/easyconfigs/r/Ray-assembler/Ray-assembler-2.3.1-iimpi-2019a.eb:6]: Ah, thanks for the notice :)

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
node2622.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/77a2ae2ff0f4121a0e38e35da3314898 for a full test report.

File: easybuild/easyconfigs/r/Ray-assembler/Ray-assembler-2.3.1-iimpi-2019a.eb [hunk 4]
```diff
@@ -15,7 +15,7 @@ toolchain = {'name': 'iimpi', 'version': '2019a'}
 toolchainopts = {'usempi': True}
 
 source_urls = [('https://sourceforge.net/projects/denovoassembler/files/', 'download')]
-sources = [SOURCE_TAR_BZ2]
+sources = ['ray-%(version)s.tar.bz2']
```
Comment (boegel) [inline easybuild/easyconfigs/r/Ray-assembler/Ray-assembler-2.3.1-iimpi-2019a.eb:18]: @Darkless012 This should be `Ray-%(version)s.tar.bz2`?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/r/Ray-assembler/Ray-assembler-2.3.1-iimpi-2019a.eb (hunk 4 above)
Comment (Darkless012) [reply easybuild/easyconfigs/r/Ray-assembler/Ray-assembler-2.3.1-iimpi-2019a.eb]: Thanks for pointing out

**Comment by lexming** [comment]
@Darkless012 thanks for fixing the name collision. Following the comments of previous reviewers I have remove the change to archived easyconfigs. I hope that you agree with this change.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node381.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/7aa86578f2a5e0c2a1b8489dabc8357f for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
node126.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/5b7e7554a438e1fa78d06de716ac9616 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2720.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/a25e53e8ee730033054cdc35b1d62c38 for a full test report.

**Pull request merged**